backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/DocumentationUnitService.java

```diff
@@ -488,6 +490,30 @@ private void saveForRecycling(Documentable documentationUnit) {
     }
   }
 
+  @Transactional(rollbackFor = BadRequestException.class)
+  public void bulkAssignProcedure(
+      @NotNull List<UUID> documentationUnitIds, String procedureLabel, User user)
+      throws DocumentationUnitNotExistsException, BadRequestException {
+    Procedure procedure = Procedure.builder().label(procedureLabel).build();
+    for (UUID documentationUnitId : documentationUnitIds) {
+      Documentable documentable = repository.findByUuid(documentationUnitId, user);
```

Comment:
potential nullpointer?
```
if (documentable == null) {
    throw new DocumentationUnitNotExistsException(...);
}
```

Reply:
`findByUUID` schmeißt direkt eine NotExists exception, deshalb bubbeled das dann einfach nach oben. Die service Methode hier hat die Exception auch in der Signatur `throws DocumentationUnitNotExistsException`. Im Controller wird daraus dann eine 400.